File: Python/18258_큐2.py
import sys
from collections import deque

if __name__ == '__main__':
	N = int(sys.stdin.readline())
	
	q = deque()
	for _ in range(N):
		arr = sys.stdin.readline().split()
		if arr[0] == 'push': # push 명령어
			q.append(int(arr[1])) # q에 해당 정수 입력
		elif arr[0] == 'pop': # pop 명령어
			if q: # q가 비어있지 않다면
				print(q.popleft()) # pop 진행(python은 pop할 때 pop한 값도 함께 return됨)
			else: # 비어있다면
				print(-1) # -1 출력
		elif arr[0] == 'size': # size 명령어
			print(len(q)) # 길이 출력(q의 길이가 q의 size가 됨)
		elif arr[0] == 'empty': # empty 명령어
			if q: # q가 비어있지 않다면
				print(0) # 0 출력
			else: # 비어있다면
				print(1) # 1 출력
		elif arr[0] == 'front': # front 명령어
			if q: # q가 비어있지 않다면
				print(q[0]) # q의 맨 앞 값 출력
			else: # 비어있다면
				print(-1) # -1 출력
		elif arr[0] == 'back': # back 명령어
			if q: # q가 비어있지 않다면
				print(q[-1]) # q의 맨 뒷 값 출력
			else: # 비어있다면
				print(-1) # -1 출력

File: C++/18258_큐2.cpp
#include <iostream>
#include <deque>
using namespace std;

int main() {
	ios::sync_with_stdio(0);
    cin.tie(0); // cin, cout 속도를 더 빠르게 해주기 위함
    
	int N;
	cin >> N;
	
	deque<int> q; // deque 사용하는 이유는 back을 출력하기 위함
	for(int i = 0; i < N; i++){
		string order;
		cin >> order; // 명령어 받기 위함
		
		if(order == "push"){ // push 명령어
			int n;
			cin >> n;
			q.push_back(n); // q에 숫자 n삽입
		}
		
		else if(order == "pop"){ // pop 명령어
			if(q.empty()){ // q가 비어있으면
				cout << "-1\n"; // -1 출력
			}
			else{ // 비어있지 않으면
				cout << q.front() << '\n';
				q.pop_front(); // pop한 문자 출력(c++에서는 pop했을 때의 return 값이 void이므로 pop되는 문자를 출력하려면 front 값을 출력한 후 pop 진행해야 함)
			}
		}
		
		else if(order == "size"){ // size 명령어
			cout << q.size() << '\n'; // q의 크기 출력
		}
			
		else if(order == "empty"){ // empty 명령어
			if(q.empty()){ // 비어있으면
				cout << "1\n"; // 1 출력
			}
			else{ // 아니면
				cout << "0\n"; // 0 출력
			}
		}
		
		else if(order == "front"){ // front 명령어
			if(q.empty()){ // q가 비어있으면
				cout << "-1\n"; // -1 출력
			}
			else{ // 아니면
				cout << q.front() << '\n'; // q의 맨 앞 값 출력
			}
		}
		
		else if(order == "back"){ // back 명령어
			if(q.empty()){ // q가 비어있으면
				cout << "-1\n"; // -1 출력
			}
			else{
				cout << q.back() << '\n'; // q의 맨 뒷 값 출력
			}
		}
	}
	
	return 0;
}
